e2e/test/provisioning/ProvisioningE2ETests.cs

```diff
@@ -1096,14 +1102,17 @@ private static async Task ConfirmExpectedDeviceCapabilitiesAsync(DeviceRegistrat
                     switch (enrollmentType)
                     {
                         case EnrollmentType.Individual:
-                            X509Certificate2Generator.GenerateSelfSignedCertificates(registrationId, s_selfSignedCertificatesFolder, Logger);
+                            X509Certificate2Helper.GenerateSelfSignedCertificateFiles(registrationId, s_certificatePassword, s_x509CertificatesFolder, Logger);
 
 #pragma warning disable CA2000 // Dispose objects before losing scope
                             // This certificate is used for authentication with IoT hub, it is disposed at the end of the test method.
-                            X509Certificate2 publicPrivateCertificate = CreateX509CertificateWithPublicPrivateKey(registrationId);
+                            certificate = X509Certificate2Helper.CreateX509Certificate2FromPfxFile(
```

Comment:
reword: generated certificate passed to the caller and will be disposed when the caller is disposed.